pkg/optimizer/datapath_csp.go

```diff
@@ -999,7 +998,7 @@ func arrayOfSameInt(num, arrayLen int) []string {
 
 func getAssetInterface(connection *datacatalog.GetAssetResponse) taxonomy.Interface {
 	if connection == nil || connection.Details.Connection.Name == "" {
-		return taxonomy.Interface{Protocol: utils.GetDefaultConnectionType(), DataFormat: ""}
+		return taxonomy.Interface{Protocol: "", DataFormat: ""}
```

Comment:
This is not going to work, as the code currently does not handle `Protocol == ""` as a special case.
What should be the semantics? Should `""` match any protocol?

Reply:
Yes, the semantics is to skip the check. This is the use-case of writing a new asset when the destination is not known in advance but is rather dictated by the module capabilities and available storage.

Reply:
Then this requires a few more check. For now, just replace `utils.GetDefaultConnectionType()` with `utils.S3`

Reply:
OK